File: src/main/java/hilfling/backend/hilfling/model/PhotoTag.java
package hilfling.backend.hilfling.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@Table(name = "photo_tag")
public class PhotoTag implements Serializable, BaseModel {
    public PhotoTag() {
    }

    public PhotoTag(String tag) {
        this.tag = tag;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tag", nullable = false, unique = true)
    private String tag;
}

File: src/main/kotlin/hilfling/backend/hilfling/model/PhotoTag.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface PhotoTag : BaseModel<PhotoTag> {
    companion object : Entity.Factory<PhotoTag>()

    var name: String
}

object PhotoTags : BaseTable<PhotoTag>("photo_tag") {
    val name = varchar("name").bindTo { it.name }
}

val Database.photo_tags get() = this.sequenceOf(PhotoTags)